### ISSUE TITLE
$500 Bounty: OHLC Candlestick Charts

### ISSUE THREAD

**Issue by hanyoonLDA** [Kozea/pygal#426]
I love pygal and how easily it integrates within our systems -- so we are offering a $500 bounty to anyone that can make OHLC candlestick charts a part of pygal! [This](http://www.pygal.org/en/latest/documentation/types/box.html) would be a good starting point, I'd assume.

**Comment by kingzley** [comment]
Is this still on?
Has anyone won the bounty yet?

**Comment by hanyoonLDA** [comment]
@1beb @kingzley If you guys are seriously interested in the bounty, let me know. Sorry about the late response, my github notifications get filled up way too quickly.

**Comment by 1beb** [comment]
@hanyoonLDA yes, interested. Please provide an example chart for reference. Specify variation between 300-500 in terms of comprehensiveness.

**Comment by qwldcl-del** [comment]
/claim

I can implement OHLC candlestick charts in pygal! Let me work on this feature.

**Comment by zhaog100** [comment]
Hi! I've implemented OHLC Candlestick Chart support in PR #579.

## Implementation Overview

The PR adds a new `Candlestick` chart type to pygal that accepts OHLC data as 4-tuples `(open, high, low, close)`.

### Key Features
- **Bullish/bearish colors**: Green for up candles (close >= open), red for down candles
- **Wick lines**: Vertical lines showing the high-low range
- **Open/close ticks**: Horizontal marks at open (left side) and close (right side)
- **Configurable**: Custom colors, fill modes (`up_down`), toggle tick visibility
- **Doji support**: Handles open == close correctly
- **Full tooltip**: Shows Open, High, Low, Close values

### Config Options
- `bullish_color` / `bearish_color`: Customize candle colors
- `show_open` / `show_close`: Toggle price tick marks
- `up_down`: `'both'` / `'up'` / `'down'` to control fill behavior

### Files Changed
- `pygal/graph/candlestick.py` (new) - Candlestick chart implementation
- `pygal/__init__.py` - Export Candlestick
- `pygal/config.py` - Config keys for candlestick options
- `pygal/test/test_candlestick.py` (new) - 11 test functions

Looking forward to your review! 📊

**Comment by zhaog100** [comment]
Claiming this bounty. I'll implement OHLC candlestick chart support.

**Comment by 1beb** [comment]
Watching the AIs fight over a 7 year old bounty is hilarious.

**Comment by jamilahmadzai** [comment]
Hi @LPX55 and pygal maintainers, before I start any implementation work here, can you confirm whether this 2018 bounty is still active and payable?

The points I need to confirm first are:

- whether the payout is still available, and whether the expected amount is $300 or up to $500
- what payout rail should be used if the work is accepted
- who can validate the bounty, since the current open PR #579 already implements OHLC/candlestick support and one maintainer noted they cannot validate the feature or bounty
- whether a fresh PR would be useful, or whether review should focus on the existing #579 path instead

I do not want to open a duplicate implementation unless the bounty and desired direction are current.

**Comment by glepretre** [comment]
Thank you to everyone who has taken an interest in this issue.

We would like to clarify that the pygal project has never had a bounty program, nor any policy supporting third-party bounties. Any reward that may have been offered in relation to this issue was proposed solely by the original author (@LPX55) and does not involve the pygal maintainers or Kozea in any way.

Unfortunately, this issue has recently attracted a large number of pull requests, many of them generated by AI agents. Several of these submissions were clearly incorrect, unrelated to the requested feature, or even destructive (for example, proposing changes that removed large portions of the project). Reviewing such contributions consumes significant maintainer time and does not help move the project forward.

Open source projects depend on the work of human contributors and maintainers who understand the codebase, the project's goals, and its long-term maintenance requirements. The recent rise of AI-generated contributions has increased the volume of submissions, but it has not reduced the need for careful human review and engineering judgment.

As a result, we are closing this issue. Future project guidelines will explicitly prohibit the use of bounties for pygal-related development and contributions.

If pygal continues to receive large numbers of low-quality or AI-generated pull requests, we may have no choice but to place the repository in read-only mode to protect the maintainers' ability to manage the project effectively.

Thank you to those who have contributed constructively and in good faith.

**Comment by LPX55** [comment]
Please see my comment on #579 - thanks all, and apologies for the confusion.
